docs/decisions/0008-migrating-to-new-learner-credit.md

```diff
@@ -0,0 +1,159 @@
+
+To mitigate this concern, we will deprecate the existing `CourseRunCards` component in favor of creating a more streamlined, lightweight `CourseRunCards` component instead. That is, we will have `CourseRunCards` that is integrated with the EMET APIs and fallback to `CourseRunCards.Deprecated` to remain backwards compatible.
+
+#### Triggering EMET redemption for a course run
+When a learner clicks on the "Enroll" CTA for a course run that is redeemable by the EMET system (i.e., has learner credit enabled with balance remaining), the general logic is as follows:
+* Make a `POST` request to the `redeem` endpoint for the redeemable access policy returned by `can_redeem`. This returns the transaction payload from `enterprise-subsidy`.
+* Poll against `enterprise-subsidy` API for the status of the returned transaction UUID until the transaction is in a non-pending state (e.g., "committed").
+  * If in a "committed" state, redirect learner to courseware URL for OCM courses to begin learning.
+  * If in an "error" state, we will ensure appropriate messaging is displayed and an option to retry.
+
+#### Backwards compatibility with non-EMET subsidy types
+
+The proposed logic (subject to change at implementation) for the course page to attempt to redeem with the EMET APIs but remain backwards compatible with subscription licenses, legacy enterprise offers, and coupon codes is as follows:
+
+1. Learner lands on course page.
+1. If `FEATURE_ENABLE_EMET_REDEMPTION` is enabled:
+    * Fetch `can_redeem` API from `enterprise-access`. This tells us whether the learner can attempt to redeem the course based on the state of subsidies available to the enterprise/learner in `enterprise-subsidy` / `enterprise-access`.
+    * If `can_redeem` returns a redeemable access policy, we will use the new/simpler `CourseRunCards` component (and a EMET-integrated `StatefulEnroll` component via the new `CourseRunCard`).
+    * If `can_redeem` does not return a redeemable access policy for the learner/course, stick to using `CourseRunCards.Deprecated` to fallback to current state (which supports subscriptions, coupon codes, etc.).
+1. If `FEATURE_ENABLE_EMET_REDEMPTION` is NOT enabled:
+    * Render `CourseRunCards.Deprecated` to stick to current state.
+
+By implementing this "smart" fallback logic, we will ensure that the course page remains backwards compatible with non-EMET subsidy types (e.g., subscription licenses).
+
+In order to support existing enrollments that were redeemed outside of the EMET system (e.g., subscription license), the new `CourseRunCard` component (via `CourseRunCards`) will continue to rely on parsing the learner's existing `EnterpriseCourseEnrollments`, data available and used by the course page today, to understand whether the learner has an existing enterprise enrollment that was subsidized outside of the EMET system.
+
+## Consequences
```

Comment:
One other consequence related to the point made in the rejected alternatives:

* This results in enterprise business logic embedded in frontend code for as long as it takes for us to migrate or age-off old subsidies.

Reply:
Good callout, will add!